public/static/js/subscriptions.js

```diff
@@ -266,11 +266,17 @@ function subCtrl($scope, $mdDialog, mciUserSettingsService) {
           return;
         }
         $scope.bindTrigger();
+        // The subscription's trigger_data fields are now set to their default values (if a default value is declared).
         _.each($scope.c.subscription.regex_selectors, function(selector){
           var typeLabel = $scope.findRegexTypeLabel(selector.type);
           $scope.regexSelectors[selector.type] = {type_label: typeLabel.type_label, data: selector.data};
         });
-        $scope.extraData = $scope.c.subscription.trigger_data;
+        // If there is an explicit value for a trigger_data field, its default value is overridden.
+        if($scope.c.subscription.trigger_data) {
```

Comment:
I don't think this fixes the problem does it? $scope.extraData will still be undefined. If this is due to data being saved in an old format, can we default in values if undefined?

Reply:
As discussed in person